src/OpenTelemetry.AutoInstrumentation/AutoInstrumentationEventSource.cs

```diff
@@ -0,0 +1,46 @@
+// <copyright file="AutoInstrumentationEventSource.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics.Tracing;
+
+namespace OpenTelemetry.AutoInstrumentation;
+
+/// <summary>
+/// EventSource implementation for OpenTelemetry AutoInstrumentation implementation.
+/// </summary>
+[EventSource(Name = "OpenTelemetry-AutoInstrumentation")]
+internal class AutoInstrumentationEventSource : EventSource
```

Comment:
Please add private constructor, to avoid multiple instances of this class.
We have had CI issue related to this stuff: #1230

Reply:
This is for EventSource data collection, we won't be able to mark this as lazy. This is being utilized by framework itself and initialized by us. 

Only with EventListener  we might run into multiple instance issue.